ArgoServer/src/server/Server.cpp

```diff
@@ -0,0 +1,180 @@
+﻿#include "stdafx.h"
+#include "Server.h"
+
+/// <summary>
+/// Constructor for server,
+/// will call initialization of server
+/// </summary>
+/// <param name="_port">port to open the server on</param>
+app::net::Server::Server(int _port)
+{
+	initServer(_port);
+}
+
+/// <summary>
+/// Method will listen for sockets connecting as well as check if any sockets sent anything.
+/// 
+/// </summary>
+bool app::net::Server::listenForSockets()
+{
+	bool got_socket = acceptSocket(totalConnections);
+	if (!got_socket)
+	{
+		//app::Console::writeLine({ "Failed to accept the clients connection" });
+		return false;
+	}
+	else
+	{
+		app::Console::writeLine({ "Client connected! ID: ", std::to_string(totalConnections) });
+		CreateThread(NULL, NULL, (LPTHREAD_START_ROUTINE)clientHandlerThread, (LPVOID)(totalConnections), NULL, NULL);
+		totalConnections += 1;
+		return true;
+	}
+}
+
+
+/// <summary>
+/// Method to initialize the server.
+/// creates server at passed port
+/// </summary>
+/// <param name="_port">port to connect to</param>
+void app::net::Server::initServer(int _port)
+{
+	//Initialize SDL stuff
+	if (SDL_Init(SDL_INIT_EVERYTHING) != NULL)
+	{
+		app::Console::writeLine({ "ERROR: Initializing SDL [", SDL_GetError(), "]" });
+	}
+	if (SDLNet_Init() != NULL)
+	{
+		app::Console::writeLine({ "ERROR: Initializing SDLNet [", SDLNet_GetError(), "]" });
+	}
+
+	//start up the server
+	IPaddress ip;
+	//attempt to resolve the host
+	if (SDLNet_ResolveHost(&ip, NULL, 27000) == -1)
+	{
+		app::Console::writeLine({ "ERROR: SDLNet_ResolveHost: [", SDLNet_GetError(), "]" });
+		exit(EXIT_FAILURE);
+	}
+	if (SDLNet_ResolveIP(&ip) == NULL) {
+		printf("SDLNet_ResolveIP: %s\n", SDLNet_GetError());
+		exit(1);
+	}
+	// Get our IP address in proper dot-quad format by breaking up the 32-bit unsigned host address and splitting it into an array of four 8-bit unsigned numbers...
```

Comment:
Write this into a member function, as it has nothing to do with initializing server.
```cpp
app::net::Server::output(whatever_type_(ip.host)_is const & host)
```